Adding Polly recording info, update to SDK 8 (#37)
JupiterOne/integration-template: test/README.md

> ## Notes on test/config.ts

Reviewer: It would be good to provide a brief overview of testing the integration in README.md with a link to these details about the `test/config.ts`.

Author: Good thought. Did that.

> Config.ts is designed to support the ability to run tests with local variables
> stored in the .env file, or with dummy variables on Github CI/CD. During
> development, run `yarn test:env` to execute tests using the variables in the
> .env file on your local machine.

Reviewer: I think it is good to reference the file consistently as `test/config.ts`.

```suggestion
`test/config.ts` is designed to support the ability to run tests with the environment
variables stored in the `.env` file, or with dummy variables in CI/CD. During
development, run `yarn test:env` to execute tests using the variables in the
`.env` file on your local machine.
```

Author: Fixed.

> To see the tests run as they will on Github, run `yarn test:ci`. Tests will run
> using the dummy variables in config.ts. Authentication will fail unless
> recordings are in place.

Reviewer: ```suggestion
To see the tests run as they will during CI/CD, run `yarn test:ci`. Tests will run
using the dummy variables in `test/config.ts`. Authentication will fail unless
recordings are in place.
```

Author: Fixed.

## Notes on test/recording.ts

Recording.ts is designed to make recordings of the API responses using Polly, so
that CI/CD tests using dummy variables can pass. When the tests run, if there is
no recording, Polly records the API responses, mutates them according to

> parameters in recording.ts, and then saves that file. In the future, if the same

Reviewer: It's not entirely clear what "that file" refers to here.

Author: Fixed

version of a more sophisticated redaction routine in `test/recording.ts`.

Once recording is enabled, but _BEFORE_ committing the recording files via git,
examine the recording file(s) for secrets. Then modify `test/recording.ts`
accordingly, delete the recording file(s), and record again. Repeat until you
have a clean recording to commit to git.

If you commit a dirty recording, your git history will still have the secrets
even if you fix them in a later commit. Bad actors know this and regularly scan
git history. Therefore, if you acccidentally commit a dirty recording, it will
require modification of your git history.

## Advanced Polly options

Polly matches a URL request to a recording via a guid. By default, the guid is
generated considering the whole URL (including for example any query parameters)
plus the header properties sent and how many times that URL has been called in
this recording session.

For many situations, this will work just fine, but there are occasions where you
need to modify what Polly looks at in generating the guid. You can do this by
passing Polly options to your recording setup function in `test/recording.ts`.
For example:

> `setupRecording({ ...input, redactedRequestHeaders: ['Authorization'], redactedResponseHeaders: ['set-cookie'], mutateEntry: (entry) => { redact(entry); }, options: { matchRequestsBy: { headers: false, order: false, }, }, });`

Reviewer: These can use three backticks, followed by the language, to get excellent formatting and code highlighting. https://github.github.com/gfm/#fenced-code-blocks

For example:

~~~~~md
```ts
setupRecording({
  ...input,
  redactedRequestHeaders: ["Authorization"],
  redactedResponseHeaders: ["set-cookie"],
  mutateEntry: (entry) => {
    redact(entry);
  },
  options: { matchRequestsBy: { headers: false, order: false } },
});
```
~~~~~

```ts
setupRecording({
  ...input,
  redactedRequestHeaders: ["Authorization"],
  redactedResponseHeaders: ["set-cookie"],
  mutateEntry: (entry) => {
    redact(entry);
  },
  options: { matchRequestsBy: { headers: false, order: false } },
});
```
